engine/lib/notification.php

```diff
@@ -504,7 +505,24 @@ function notify_user($to, $from, $subject, $message, array $params = array(), $m
 							continue;
 						}
 						$language = $recipient->language;
+
+						$params['method'] = $method;
+						$params['language'] = $language;
+						$params['sender'] = $sender;
+						$params['recipient'] = $recipient;
+
 						$notification = new \Elgg\Notifications\Notification($sender, $recipient, $language, $subject, $message, $summary, $params);
```

Comment:
The constructor $params argument receives a bunch of new keys here. Is that intentional? If not I guess I'd prefer to not pass extraneous stuff into APIs. Like `$hook_params = $params` and use that for the new keys.